### ISSUE TITLE
Internal Cache$Key is not Serializable

### ISSUE THREAD

**Issue by piyushrai** [jOOQ/jOOQ#5290]
Trying to cache JOOQ record in redis and getting following exception

org.springframework.data.redis.serializer.SerializationException: Cannot serialize; nested exception is org.springframework.core.serializer.support.SerializationFailedException: Failed to serialize object using DefaultSerializer; nested exception is java.io.NotSerializableException: org.jooq.impl.Utils$Cache$Key

----
See also: 
http://stackoverflow.com/q/37387646/521799

**Comment by lukaseder** [comment]
Thank you very much for reporting. We'll look into this as soon as possible.

In the meantime, I can see two workarounds:

- You explicitly "detach" your records prior to serialisation in redis by calling `Record.detach()`
- You turn off automatic attaching of records to the creating `Configuration` using the `Settings.attachRecords` property

In both cases, should you wish to retrieve a record from redis and store it again, you will need to explicitly call `Record.attach(Configuration)`

**Comment by lukaseder** [comment]
Thanks again for reporting. This is now fixed for 3.9.0 and the fix will be backported to 3.8.2, 3.7.5, and 3.6.5

**Comment by piyushrai** [comment]
Thanks for the quick fix. Will this be published to the maven repository?

**Comment by lukaseder** [comment]
Yes, of course. There will be patch releases published within the next month.
